main/src/ca/uwaterloo/flix/api/lsp/provider/CompletionProvider.scala

```diff
@@ -217,7 +217,7 @@ object CompletionProvider {
       case WeederError.UndefinedAnnotation(_, _) => (1, SyntacticContext.Decl.Module)
       case err: ResolutionError.UndefinedJvmStaticField => (1, SyntacticContext.Expr.StaticFieldOrMethod(err))
       case err: TypeError.MethodNotFound => (1, SyntacticContext.Expr.InvokeMethod(err.tpe, err.methodName))
-      case err: TypeError.FieldNotFound => (1, SyntacticContext.Expr.InvokeMethod(err.tpe, err.fieldName))
+      case err: TypeError.FieldNotFound => (1, SyntacticContext.Expr.AccessField(err.tpe, err.fieldName))
```

Comment:
You don't need to extend SyntacticContext.

Instead, try to extend the getSemanticCompletions function.